synedrion/src/paillier/keys.rs

```diff
@@ -116,31 +151,37 @@ impl<P: PaillierParams> SecretKeyPaillierPrecomputed<P> {
         self.sk.clone()
     }
 
-    pub fn primes(&self) -> (Signed<P::Uint>, Signed<P::Uint>) {
+    #[allow(clippy::type_complexity)]
+    pub fn primes(&self) -> (SecretBox<Signed<P::Uint>>, SecretBox<Signed<P::Uint>>) {
         // The primes are positive, but where this method is used Signed is needed,
         // so we return that for convenience.
-        // TODO (#77): must be wrapped in a Secret
         (
-            Signed::new_positive(self.sk.p.into_wide(), P::PRIME_BITS as u32).unwrap(),
-            Signed::new_positive(self.sk.q.into_wide(), P::PRIME_BITS as u32).unwrap(),
+            Signed::new_positive(self.sk.p.expose_secret().into_wide(), P::PRIME_BITS as u32)
+                .expect("The primes in the `SecretKeyPaillier` are 'safe primes' and positive by construction; the bound is assumed to be configured correctly by the user.")
+                .secret_box(),
+            Signed::new_positive(self.sk.q.expose_secret().into_wide(), P::PRIME_BITS as u32)
+                .expect("The primes in the `SecretKeyPaillier` are 'safe primes' and positive by construction; the bound is assumed to be configured correctly by the user.")
+                .secret_box(),
         )
     }
 
-    pub fn totient(&self) -> &Bounded<P::Uint> {
-        // TODO (#77): must be wrapped in a Secret
-        &self.totient
+    /// Returns Euler's totient function (`φ(n)`) of the modulus, wrapped in a [`SecretBox`].
+    pub fn totient(&self) -> SecretBox<Bounded<P::Uint>> {
+        self.totient.into()
     }
 
-    /// Returns Euler's totient function of the modulus.
+    /// Returns Euler's totient function (`φ(n)`) of the modulus as a [`NonZero`]
     pub fn totient_nonzero(&self) -> NonZero<P::Uint> {
-        // TODO (#77): must be wrapped in a Secret
-        NonZero::new(*self.totient.as_ref()).unwrap()
+        // TODO (#77): must be wrapped in a Secret – This must await the release of crypto-bigint
```

Comment:
Is it still the case in 0.6.0-rc3?

Reply:
In v0.6 it is implemented but this PR is against v0.5 so it'll have to wait. :/